sdks/java/io/hadoop-input-format/src/test/java/org/apache/beam/sdk/io/hadoop/inputformat/integration/tests/HIFIOCassandraIT.java

```diff
@@ -0,0 +1,146 @@
+ * "--serverPort=<port>" ]' -Dit.test=HIFIOCassandraIT -DskipITs=false
+ *
+ */
+@RunWith(JUnit4.class)
+@FixMethodOrder(MethodSorters.JVM)
+public class HIFIOCassandraIT implements Serializable {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(HIFIOCassandraIT.class);
+  private static final String CASSANDRA_KEYSPACE = "beamdb";
+  private static final String CASSANDRA_TABLE = "scientists";
+  private static HIFTestOptions options;
+
+  @BeforeClass
+  public static void setUp() {
+    PipelineOptionsFactory.register(HIFTestOptions.class);
+    options = TestPipeline.testingPipelineOptions().as(HIFTestOptions.class);
+    LOGGER.info("Pipeline created successfully with the options.");
+  }
+
+  /**
+   * This test reads data from the Cassandra instance and verifies if data is read successfully.
+   *
+   */
+  @Test
+  public void testHIFReadForCassandra() {
+    Pipeline pipeline = TestPipeline.create(options);
+    Configuration conf = getConfiguration(options);
+    SimpleFunction<Row, String> myValueTranslate = new SimpleFunction<Row, String>() {
+      @Override
+      public String apply(Row input) {
+        return input.getString("scientist");
+      }
+    };
+    PCollection<KV<Long, String>> cassandraData =
+        pipeline.apply(HadoopInputFormatIO.<Long, String>read().withConfiguration(conf)
+            .withValueTranslation(myValueTranslate));
+    PAssert.thatSingleton(cassandraData.apply("Count", Count.<KV<Long, String>>globally()))
+        .isEqualTo(10L);
+
+    List<KV<Long, String>> expectedResults =
+        Arrays.asList(KV.of(1L, "Faraday"), KV.of(2L, "Newton"), KV.of(3L, "Galilei"),
+            KV.of(4L, "Maxwell"), KV.of(5L, "Pasteur"), KV.of(6L, "Copernicus"),
+            KV.of(7L, "Curie"), KV.of(8L, "Bohr"), KV.of(9L, "Darwin"), KV.of(10L, "Einstein"));
+    PAssert.that(cassandraData).containsInAnyOrder(expectedResults);
+    pipeline.run();
+  }
+
+  /**
+   * This test reads data from the Cassandra instance based on query and verifies if data is read
+   * successfully.
+   *
+   */
+  @Test
+  public void testHIFReadForCassandraQuery() {
+    Pipeline pipeline = TestPipeline.create(options);
+    Configuration conf = getConfiguration(options);
+    conf.set("cassandra.input.cql", "select * from " + CASSANDRA_KEYSPACE + "." + CASSANDRA_TABLE
+        + " where token(id) > ? and token(id) <= ? and scientist='Einstein' allow filtering");
+    SimpleFunction<Row, String> myValueTranslate = new SimpleFunction<Row, String>() {
+      @Override
+      public String apply(Row input) {
+        return input.getString("id");
+      }
+    };
+    PCollection<KV<Long, String>> cassandraData =
+        pipeline.apply(HadoopInputFormatIO.<Long, String>read().withConfiguration(conf)
+            .withValueTranslation(myValueTranslate));
+    PAssert.thatSingleton(cassandraData.apply("Count", Count.<KV<Long, String>>globally()))
+        .isEqualTo(1L);
```

Comment:
Yes, we can validate the id. But as per our discussion, we are trying to implement this validation through checksum. Once that is done, we will add similar checksum validation for id here.